torch/csrc/jit/script/compiler.h

```diff
@@ -35,6 +35,12 @@ struct SugaredValue : public std::enable_shared_from_this<SugaredValue> {
     throw ErrorReport(loc) << "attribute lookup is not defined on " << kind();
   }
 
+  // use it as a vector of values, e.g. a tuple of values as return value from
+  // a method invocation
+  virtual const std::vector<Value*>& asValues(SourceRange loc, Method& m) {
+    throw ErrorReport(loc) << kind() << " cannot be used as repeated values";
+  }
```

Comment:
Would be good to sync up this one with Zach's PR, because if we have statically typed tuples, then we could use them in `for` loops too. We might want to avoid having two different methods in `SugaredValues` for this